tools/github-event-processor/YmlAndConfigFiles/event-processor.yml

```diff
@@ -0,0 +1,95 @@
+  issue_comment:
+    types: [created]
+  # synchronize is the pull_request_target event when changes are pushed
+  # pull request merged is the closed event with github.event.pull_request.merged = true
+  pull_request_target:
+    types: [closed, labeled, opened, reopened, review_requested, synchronize, unlabeled]
+  pull_request_review:
+    types: [submitted]
+
+# This removes all unnecessary permissions, the ones needed will be set below.
+# https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
+permissions: {}
+
+jobs:
+  event-handler:
+    permissions:
+      issues: write
+      pull-requests: write
+      # For OIDC auth
+      id-token: write
+      contents: read
+    name: Handle ${{ github.event_name }} ${{ github.event.action }} event
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: 'Az CLI login'
+        if: ${{ github.event_name == 'issues' && github.event.action == 'opened' }}
+        uses: azure/login@v1
+        with:
+            client-id: ${{ secrets.AZURE_CLIENT_ID }}
+            tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+            subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: 'Run Azure CLI commands'
+        if: ${{ github.event_name == 'issues' && github.event.action == 'opened' }}
+        run: |
+          LABEL_SERVICE_API_KEY=$(az keyvault secret show \
+            --vault-name issue-labeler \
+            -n issue-labeler-func-key \
+            -o tsv \
+            --query value)
+
+          echo "::add-mask::$LABEL_SERVICE_API_KEY"
+          echo "LABEL_SERVICE_API_KEY=$LABEL_SERVICE_API_KEY" >> $GITHUB_ENV
+
+      # To run github-event-processor built from source, for testing purposes, uncomment everything
+      # in between the Start/End-Build From Source comments and comment everything in between the
+      # Start/End-Install comments
+      # Start-Install
+      - name: Install GitHub Event Processor
+        run: >
+          dotnet tool install
+          Azure.Sdk.Tools.GitHubEventProcessor
+          --version 1.0.0-dev.20230303.6
+          --add-source https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json
+          --global
+        shell: bash
+
+      - name: Process Action Event
+        run: |
+          echo $GITHUB_PAYLOAD > payload.json
+          github-event-processor ${{ github.event_name }} payload.json
+        shell: bash
+        env:
+          GITHUB_PAYLOAD: ${{ toJson(github.event) }}
+          # This is a temporary secret generated by github
+          # https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          LABEL_SERVICE_API_KEY: ${{ env.LABEL_SERVICE_API_KEY }}
+      # End-Install
+
+      # To run github-event-processor built from source, for testing purposes, uncomment this
```

Comment:
I know you haven't done it yet but I think we still want this to clone down the tools repo as part of this step.

Reply:
> I know you haven't done it yet but I think we still want this to clone down the tools repo as part of this step.

That's going to be the plan just as soon as I have a chance to play with the [checkout action](https://github.com/actions/checkout). I believe I'm going to want to compile and install. I don't believe that the directory climber, that's looking for the config and the CODEOWNERS files, is going to find the right ones for the repository unless I install what's built and run it globally.

Reply:
[Commit 12](https://github.com/Azure/azure-sdk-tools/pull/5607/commits/ab62fad56e35c5ae6e167dc620bc26be82252c95) on this PR has this change. It uses the checkout action to pull the azure-sdk-tools repository and then builds, packs and installs what it just built globally. This simplifies things because only the build/install from source and install from feed need to be commented/uncommented which keeps the run commands the same in both cases.